lxd/device/unix_hotplug.go

```diff
@@ -230,27 +236,32 @@ func (d *unixHotplug) loadUnixDevice() *udev.Device {
 	}
 
 	devices, _ := e.Devices()
-	var device *udev.Device
+	var matchingDevices []*udev.Device
 	for i := range devices {
-		device = devices[i]
+		device := devices[i]
 
-		if device == nil {
+		if device == nil || device.Devnum().Major() == 0 || device.Devnode() == "" || strings.HasPrefix(device.Subsystem(), "usb") {
 			continue
 		}
 
-		devnum := device.Devnum()
-		if devnum.Major() == 0 || devnum.Minor() == 0 {
-			continue
+		match := false
+		vendorIDMatch := device.PropertyValue("ID_VENDOR_ID") == d.config["vendorid"]
```

Comment:
why is this needed when there is apparently filtering being setup via `e.AddMatchProperty` above?

Reply:
`e.AddMatchProperty` calls `udev_enumerate_add_match_property` under the hood. Each matching result is ORed - only at least one property needs to match.

Reply:
From the udevadm man page:

```
-p, --property-match=PROPERTY=VALUE
           Trigger events for devices with a matching property value. This option supports shell style pattern matching. When this option is specified more than once, then each matching result is
           ORed, that is, devices which have one of the specified properties are triggered.
```

I've removed the redundant `e.AddMatchProperty` calls since we have the new consolidated matching logic now.

Reply:
Ok i see. The AddMatchProperty calls should remain though to provide early push down filtering.

Reply:
Sounds good - I've added them back.